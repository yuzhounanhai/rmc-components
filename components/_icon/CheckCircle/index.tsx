import React, {
  ComponentProps,
} from 'react';
import Icon from '@ant-design/icons';

const SVGIcon = () => (
  <svg viewBox="0 0 72 72" width="1em" height="1em"><g fill="none" fillRule="evenodd"><path d="M36 72c19.882 0 36-16.118 36-36S55.882 0 36 0 0 16.118 0 36s16.118 36 36 36zm0-2c18.778 0 34-15.222 34-34S54.778 2 36 2 2 17.222 2 36s15.222 34 34 34z" fill="#FFF"></path><path stroke="#FFF" strokeWidth="2" d="M19 34.54l11.545 11.923L52.815 24"></path></g></svg>
);

export default (props: ComponentProps<typeof Icon>) => (
  <Icon
    component={SVGIcon}
    {...props}
  />
);