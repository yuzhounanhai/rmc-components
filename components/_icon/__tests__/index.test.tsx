import React from 'react';
import baseTest from '../../../tests/common/baseTest';
import CloseCircle from '../CloseCircle/index';
import CheckCircle from '../CheckCircle/index';

describe('Custom Icon', () => {
  baseTest(<CloseCircle />);
  baseTest(<CheckCircle />);
});