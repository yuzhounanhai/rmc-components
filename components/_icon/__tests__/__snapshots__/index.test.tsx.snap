// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Icon component snapshots tests 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "anticon",
      "role": "img",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "height": "1em",
          "viewBox": "0 0 72 72",
          "width": "1em",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "fill": "none",
              "fill-rule": "evenodd",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "d": "M36 72c19.882 0 36-16.118 36-36S55.882 0 36 0 0 16.118 0 36s16.118 36 36 36zm0-2c18.778 0 34-15.222 34-34S54.778 2 36 2 2 17.222 2 36s15.222 34 34 34z",
                  "fill": "#FFF",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": Node {
                  "attribs": Object {
                    "d": "M22 22l28.304 28.304m-28.304 0L50.304 22",
                    "stroke": "#FFF",
                    "stroke-width": "2",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "fill": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "fill": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "d": "M22 22l28.304 28.304m-28.304 0L50.304 22",
                  "stroke": "#FFF",
                  "stroke-width": "2",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "d": "M36 72c19.882 0 36-16.118 36-36S55.882 0 36 0 0 16.118 0 36s16.118 36 36 36zm0-2c18.778 0 34-15.222 34-34S54.778 2 36 2 2 17.222 2 36s15.222 34 34 34z",
                    "fill": "#FFF",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                    "fill": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                    "fill": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                },
              },
            ],
            "name": "g",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "fill": undefined,
              "fill-rule": undefined,
            },
            "x-attribsPrefix": Object {
              "fill": undefined,
              "fill-rule": undefined,
            },
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "height": undefined,
          "viewBox": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "height": undefined,
          "viewBox": undefined,
          "width": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "role": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "role": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Custom Icon component snapshots tests 2`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "anticon",
      "role": "img",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "height": "1em",
          "viewBox": "0 0 72 72",
          "width": "1em",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "fill": "none",
              "fill-rule": "evenodd",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "d": "M36 72c19.882 0 36-16.118 36-36S55.882 0 36 0 0 16.118 0 36s16.118 36 36 36zm0-2c18.778 0 34-15.222 34-34S54.778 2 36 2 2 17.222 2 36s15.222 34 34 34z",
                  "fill": "#FFF",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": Node {
                  "attribs": Object {
                    "d": "M19 34.54l11.545 11.923L52.815 24",
                    "stroke": "#FFF",
                    "stroke-width": "2",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "fill": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "fill": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "d": "M19 34.54l11.545 11.923L52.815 24",
                  "stroke": "#FFF",
                  "stroke-width": "2",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "d": "M36 72c19.882 0 36-16.118 36-36S55.882 0 36 0 0 16.118 0 36s16.118 36 36 36zm0-2c18.778 0 34-15.222 34-34S54.778 2 36 2 2 17.222 2 36s15.222 34 34 34z",
                    "fill": "#FFF",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                    "fill": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                    "fill": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                },
              },
            ],
            "name": "g",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "fill": undefined,
              "fill-rule": undefined,
            },
            "x-attribsPrefix": Object {
              "fill": undefined,
              "fill-rule": undefined,
            },
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "height": undefined,
          "viewBox": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "height": undefined,
          "viewBox": undefined,
          "width": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "role": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "role": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
